lnwallet/btcwallet/btcwallet.go

```diff
@@ -844,3 +844,75 @@ func (b *BtcWallet) IsSynced() (bool, int64, error) {
 
 	return true, bestTimestamp, nil
 }
+
+// GetRecoveryInfo returns a boolean indicating whether the wallet is started
+// in recovery mode. It also returns a float64, ranging from 0 to 1,
+// representing the recovery progress made so far.
+//
+// This is a part of the WalletController interface.
+func (b *BtcWallet) GetRecoveryInfo() (bool, float64, error) {
+	isRecoveryMode := true
+	progress := float64(0)
+
+	// A zero value in RecoveryWindow indicates there is no trigger of
+	// recovery mode.
+	if b.cfg.RecoveryWindow == 0 {
+		isRecoveryMode = false
+		return isRecoveryMode, progress, nil
+	}
+
+	// Query the wallet's birthday block height from db.
+	var birthdayBlock waddrmgr.BlockStamp
+	err := walletdb.View(b.db, func(tx walletdb.ReadTx) error {
+		var err error
+		addrmgrNs := tx.ReadBucket(waddrmgrNamespaceKey)
+		birthdayBlock, _, err = b.wallet.Manager.BirthdayBlock(addrmgrNs)
```

Comment:
Something I came across while testing this (easier to see with a neutrino backend):

The wallet doesn't start until the backend is synced, meaning that the birthday block won't be set until after then. We'll need to handle this gracefully, otherwise we'll run into the error: `unable to get wallet recovery info: birthday block not set`.

Reply:
good catch, perhaps just returning 0 in that case is best

Reply:
probably not the nicest fix...but the `"birthday block not set"` error is caught and compared, using the string value. And 0 is returned in that case.